test/TorchSharpTest/TestTorchTensor.cs

```diff
@@ -239,6 +231,114 @@ public void Test6DToString()
             }
         }
 
+        [Fact]
+        public void TestToString1()
+        {
+            Tensor t = torch.zeros(2, 2);
+            var str = t.ToString();
+            Assert.Equal("[2x2], type = Float32, device = cpu", str);
+        }
+
+
+        [Fact]
+        public void Test1DToNumpyString()
+        {
+            Assert.Equal("[0 0 0 0]", torch.zeros(4).str());
+            Assert.Equal("[0 0 0 0]", torch.zeros(4, torch.complex64).str());
+            {
+                Tensor t = torch.ones(4, torch.complex64);
+                for (int i = 0; i < t.shape[0]; i++) t[i] = torch.tensor((1.0f * i, 2.43f * i * 2), torch.complex64);
+                var str = t.str();
+                Assert.Equal($"[0 1+4.86i 2+9.72i 3+14.58i]", str);
+            }
+            Assert.Equal("[1 1 1 ... 1 1 1]", torch.ones(100, ScalarType.Int32).str());
+            Assert.Equal("[1 3 5 ... 195 197 199]", torch.tensor(Enumerable.Range(0, 100).Select(x => 2 * x + 1).ToList(), ScalarType.Float32).str());
+        }
+
+        [Fact]
+        public void Test2DToNumpyString()
+        {
+            Assert.Equal("[[0 3.141 6.2834 3.1415]\n [6.28e-06 -13.142 0.01 4713.1]]", torch.tensor(new float[] { 0.0f, 3.141f, 6.2834f, 3.14152f, 6.28e-06f, -13.141529f, 0.01f, 4713.14f }, 2, 4).str());
+            {
+                Tensor t = torch.zeros(5, 5, torch.complex64);
+                for(int i = 0; i < t.shape[0]; i++)
+                    for(int j = 0; j < t.shape[1]; j++)
+                        t[i][j] = torch.tensor((1.24f * i, 2.491f * i * 2), torch.complex64);
+                var str = t.str();
+                Assert.Equal("[[0 0 0 0 0]\n [1.24+4.982i 1.24+4.982i 1.24+4.982i 1.24+4.982i 1.24+4.982i]\n [2.48+9.964i 2.48+9.964i 2.48+9.964i 2.48+9.964i 2.48+9.964i]\n [3.72+14.946i 3.72+14.946i 3.72+14.946i 3.72+14.946i 3.72+14.946i]\n [4.96+19.928i 4.96+19.928i 4.96+19.928i 4.96+19.928i 4.96+19.928i]]", str);
+            }
+            Assert.Equal("[[0 0 0 0]\n [0 0 0 0]]", torch.zeros(2, 4, torch.complex64).str());
+            Assert.Equal("[[1 1 1 1]\n [1 1 1 1]]", torch.ones(2, 4, torch.complex64).str());
+            Assert.Equal("[[7.5 10 12.5 ... 125 127.5 130]\n [132.5 135 137.5 ... 250 252.5 255]]", torch.tensor(Enumerable.Range(1, 100).Select(x => x * 2.5 + 5).ToList(), new long[]{2, 50}, ScalarType.Float32).str());
+            Assert.Equal("[[7.5 10 12.5 ... 250 252.5 255]\n [257.5 260 262.5 ... 500 502.5 505]\n [507.5 510 512.5 ... 750 752.5 755]\n ...\n [24258 24260 24262 ... 24500 24502 24505]\n [24508 24510 24512 ... 24750 24752 24755]\n [24758 24760 24762 ... 25000 25002 25005]]",
```

Comment:
The '\n' is problematic -- this test is not likely to work across platforms.  I put a declaration in TestTorchTensor for this reason:

```C#
private string _sep = Environment.NewLine;
```

and used it in the 'expected' strings in the tests.